product/src/main/java/com/yas/product/service/ProductService.java

```diff
@@ -46,15 +46,12 @@ public ProductService(ProductRepository productRepository, MediaService mediaSer
         this.productImageRepository = productImageRepository;
     }
 
-    public List<ProductListVm> getProducts() {
-        return productRepository.findAll().stream()
-                .map(ProductListVm::fromModel)
-                .toList();
-    }
-
-    public ProductListGetVm getAllProducts(int pageNo, int pageSize) {
+    public ProductListGetVm getProductsByBrandOrName(int pageNo, int pageSize, String productName, String brandName) {
         Pageable pageable = PageRequest.of(pageNo, pageSize);
-        Page<Product> productPage = productRepository.findByOrderByIdAsc(pageable);
+        Page<Product> productPage = productRepository.findProductsByFilterOrSearching(
+                productName.trim().toLowerCase(),
```

Comment:
I think we don't need to call toLowerCase here?

Reply:
i want to search product by name which means ignore case. For instance: we can find "Samsung" by enter "samsung" keyword

Reply:
This should be handled by collation in database. I though it was case insensitive already

Reply:
> This should be handled by collation in database. I though it was case insensitive already

i got you. Thanks for your suggestion, i fixed the code . Please have a check